tests/serde.rs

```diff
@@ -28,10 +28,25 @@ fn assert_json<R>(lhs_value: JsValue, rhs: R)
 where
     R: Serialize + DeserializeOwned + PartialEq + Debug,
 {
-    assert_eq!(
-        js_sys::JSON::stringify(&lhs_value).unwrap(),
-        serde_json::to_string(&rhs).unwrap(),
-    );
+    if lhs_value.is_object() {
+        assert_eq!(
+            js_sys::JSON::stringify(&lhs_value).unwrap(),
+            serde_json::to_string(&rhs).unwrap(),
+        );
+    } else if lhs_value.is_string() {
```

Comment:
It's been awhile since I made this change. I'll try reverting it and get back to you on whether it was really necessary.

Reply:
Sorry it's been awhile since I responded. I've had a busy month so far.

Anyway, I looked into it, and it looks like the `lhs_value.is_string()` branch can actually be combined with the `lhs_value.is_object()` branch. However, the `lhs_value.is_undefined() || lhs_value.is_null()` is actually necessary since `js_sys::JSON::stringify(&lhs_value).unwrap()` panics when `lhs_value` is null.

I'll combine the redundant branches and make another commit

Reply:
Here it is: c41d8e0

Reply:
> is actually necessary since js_sys::JSON::stringify(&lhs_value).unwrap() panics when lhs_value is null

Ah - the problem is that `JSON.stringify(undefined)` in JavaScript returns `undefined` and not a string. However, `JSON.stringify(null)` should work fine.

Reply:
Ok, I simplified the condition to only include a separate case for `undefined` values: c326b7f.

Reply:
Sorry I missed this comment before. This update looks good!